src/map/script.c

```diff
@@ -7713,6 +7713,66 @@ static BUILDIN(countitem2)
 	return true;
 }
 
+/*==========================================
+ * countnameditem(item ID, { <Char Name / ID> })
+ * returns number of named items.
+ *------------------------------------------*/
+static BUILDIN(countnameditem)
+{
+	int count = 0;
+	int i;
+	struct item_data* id = NULL;
+	struct map_session_data *sd;
+
+	if (script_hasdata(st, 3)) {
+		if (script_isstringtype(st, 3)) {
+			// Character name was given
+			sd = script->nick2sd(st, script_getstr(st, 3));
+		} else {
+			// Character ID was given
+			sd = script->charid2sd(st, script_getnum(st, 3));
+		}
+	} else {
+		// Use RID by default if no name was provided
+		sd = script->rid2sd(st);
+	}
+
+	// Player not attached
+	if (sd == NULL) {
+		return true;
+	}
+
+	if (script_isstringtype(st, 2)) {
+		// Get item from DB via item name
+		id = itemdb->search_name(script_getstr(st, 2));
+	} else {
+		// Get item from DB via item ID
+		id = itemdb->exists(script_getnum(st, 2));
+	}
+
+	if (id == NULL) {
+		ShowError("buildin_countnameditem: Invalid item '%s'.\n", script_getstr(st, 2));  // returns string, regardless of what it was
+		script_pushint(st, 0);
+		return false;
+	}
+
+	for (i = 0; i < MAX_INVENTORY; i++) {
```

Comment:
i better define inside for ``for (int i = ...``

Reply:
Updated